src/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java

```diff
@@ -473,57 +447,107 @@ public boolean migrateAllLegacyPages(final User user, boolean respectFrontEndPer
     			}
     	return result;
     }
-    
+
     @Override
     public HTMLPageAsset migrateLegacyPage(HTMLPage legacyPage, User user, boolean respectFrontEndPermissions) throws Exception {
         Identifier legacyident=APILocator.getIdentifierAPI().find(legacyPage);
         VersionInfo vInfo=APILocator.getVersionableAPI().getVersionInfo(legacyident.getId());
```

Comment:
We shouldn't use the reference to APILocator on methods, would be preferable to use class attributes and constructors to get the API's